core/controllers/feedback_test.py

```diff
@@ -336,3 +346,283 @@ def _get_email(n):
+
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        init_state = exploration.states[exploration.init_state_name]
+        init_interaction = init_state.interaction
+        init_interaction.default_outcome.dest = exploration.init_state_name
+        exploration.add_states(['State A', 'State 2', 'State 3'])
+        exploration.states['State A'].update_interaction_id('TextInput')
+        exploration.states['State 2'].update_interaction_id('TextInput')
+        exploration.states['State 3'].update_interaction_id('TextInput')
+        exp_services._save_exploration(self.editor_id, exploration, '', [])
+        rights_manager.publish_exploration(self.editor_id, self.EXP_ID)
+        rights_manager.assign_role_for_exploration(
+            self.editor_id, self.EXP_ID, self.owner_id,
+            rights_manager.ROLE_EDITOR)
+
+        # Create suggestions.
+        self.post_json(
+            '%s/%s' % (feconf.SUGGESTION_URL_PREFIX, self.EXP_ID), {
+                'exploration_version': 3,
+                'state_name': u'State A',
+                'suggestion_content': {
+                    'type': 'text', 'value': u'new accepted suggestion for state A'},
+            }, csrf_token)
+        self.post_json(
+            '%s/%s' % (feconf.SUGGESTION_URL_PREFIX, self.EXP_ID), {
+                'exploration_version': 1,
+                'state_name': u'State 2',
+                'suggestion_content': {
+                    'type': 'text', 'value': 'some new value'},
+            }, csrf_token)
+        self.post_json(
+            '%s/%s' % (feconf.SUGGESTION_URL_PREFIX, self.EXP_ID), {
+                'exploration_version': 2,
+                'state_name': u'State 3',
+                'suggestion_content': {'type': 'text', 'value': ''},
+            }, csrf_token)
+        self.post_json(
+            '%s/%s' % (feconf.SUGGESTION_URL_PREFIX, self.EXP_ID), {
+                'exploration_version': 2,
+                'state_name': u'State A',
+                'suggestion_content': {'type': 'text', 'value': ' '},
+            }, csrf_token)
+        self.post_json(
+            '%s/%s' % (feconf.SUGGESTION_URL_PREFIX, self.EXP_ID), {
+                'exploration_version': 1,
+                'state_name': u'State 2',
+                'suggestion_content': {'type': 'text', 'value': '#!$%'},
+            }, csrf_token)
+        self.post_json(
+            '%s/%s' % (feconf.SUGGESTION_URL_PREFIX, self.EXP_ID), {
+                'exploration_version': 2,
+                'state_name': u'State 3',
+                'suggestion_content': {'type': 'text', 'value': u'Ֆݓॵক'},
+            }, csrf_token)
+        self.logout()
+
+    def _return_true(self, thread_id, exploration_id):
+        return True
+
+    def _return_false(self, thread_id, exploration_id):
+        return False
+
+    def _return_null(self, param_one, param_two, param_three, param_four):
+        return None
+
+    def _return_null(self):
```

Comment:
I'm not sure it's a good idea to define _return_null twice. Maybe use *args if you have a variable number of arguments?